crates/re_data_ui/src/item_ui.rs

```diff
@@ -244,19 +244,68 @@ pub fn cursor_interact_with_selectable(
 
 // TODO(andreas): Move elsewhere, this is not directly part of the item_ui.
 pub fn select_hovered_on_click(
+    ctx: &mut ViewerContext<'_>,
     response: &egui::Response,
-    selection_state: &mut SelectionState,
     items: &[Item],
 ) {
     if response.hovered() {
-        selection_state.set_hovered(items.iter().cloned());
+        ctx.selection_state_mut().set_hovered(items.iter().cloned());
     }
 
     if response.clicked() {
+        // Resolve to entity path if there's only a single instance.
```

Comment:
shouldn't this go for the hovering to, i.e. be applied to `items` at the start of the function?

Reply:
figured I don't want to do this slow (?) operation every frame, but agreed this is smelly. moving it around